engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/SnapshotServiceImpl.java

```diff
@@ -226,7 +226,7 @@ public SnapshotResult takeSnapshot(SnapshotInfo snap) {
     // we are taking delta snapshot
     private DataStore findSnapshotImageStore(SnapshotInfo snapshot) {
         Boolean fullSnapshot = true;
-        Object payload = snapshot.getPayload();
+        Object payload = snapshot.getFullBackup();
```

Comment:
If 'full backup' is a boolean, better to store it as that type rather than object.